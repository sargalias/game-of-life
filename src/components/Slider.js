import React from 'react';

const Slider = ({value, onChange}) => (
  <div className="slider" >
    <div>
      <label htmlFor="speed">Step speed</label>
    </div>
    <div>
      Min
      <input
        name="speed"
        type="range"
        min="2"
        max="50"
        value={value}
        onChange={onChange}
      />
      Max
    </div>
  </div>
);

export default Slider;
