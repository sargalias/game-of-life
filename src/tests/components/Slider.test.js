import React from 'react';
import { shallow } from 'enzyme';
import Slider from '../../components/Slider';

test('Slider should render correctly', () => {
  const wrapper = shallow(<Slider />);
  expect(wrapper).toMatchSnapshot();
});

test('value is set correctly by prop', () => {
  const value = "2";
  const wrapper = shallow(<Slider value={value} />);
  expect(wrapper.find('input').render().attr('value')).toBe(value);
});

test('onChange is assigned and called correctly', () => {
  const onChange = jest.fn();
  const value = "2";
  const wrapper = shallow(<Slider value={value} onChange={onChange} />);

  wrapper.find('input').simulate('change');
  expect(onChange).toHaveBeenCalled();
});
