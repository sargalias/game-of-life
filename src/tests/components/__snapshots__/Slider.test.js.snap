// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should render correctly 1`] = `
<div
  className="slider"
>
  <div>
    <label
      htmlFor="speed"
    >
      Step speed
    </label>
  </div>
  <div>
    Min
    <input
      max="50"
      min="1"
      name="speed"
      type="range"
    />
    Max
  </div>
</div>
`;
